NuRadioReco/modules/RNO_G/crRNOGTemplateCreator.py

```diff
@@ -0,0 +1,248 @@
+from NuRadioReco.framework.station import Station
+from NuRadioReco.framework.channel import Channel
+from NuRadioReco.framework.sim_station import SimStation
+from NuRadioReco.framework.sim_channel import SimChannel
+from NuRadioReco.framework.parameters import stationParameters
+from NuRadioReco.framework.parameters import electricFieldParameters
+from NuRadioReco.framework.electric_field import ElectricField
+import pickle
+
+class crRNOGTemplateCreator:
+    """
+        Creates CR templates by assuming a gaussian function for the electric field
+
+    """
+
+    def __init__(self):
+        self.logger = logging.getLogger("NuRadioReco.crRNOGTemplateCreator")
+        self.__detector_file = None
+
+        self.__template_run_id = None
+        self.__template_channel_id = None
+        self.__template_station_id = None
+
+        self.__sampling_rate = None
+        self.__template_sample_number = None
+
+        self.__antenna_rotation = None
+        self.__Efield_width = None
+        self.__Efield_amplitudes = None
+        self.__template_event_id = None
+
+        self.__cr_zenith = None
+        self.__cr_azimuth = None
+
+        self.__debug = None
+
+        self.__template_save_path = None
+
+        self.__efieldToVoltageConverter = NuRadioReco.modules.efieldToVoltageConverter.efieldToVoltageConverter()
+        self.__hardwareResponseIncorporator = NuRadioReco.modules.RNO_G.hardwareResponseIncorporator.hardwareResponseIncorporator()
+
+    def begin(self, detector_file, template_save_path, debug=False, logger_level=logging.NOTSET):
+        """
+                begin method
+
+                Parameters
+                ----------
+                detector_file: string
+                    path to the detector file used for the template set
+                template_save_path: string
+                    path to a folder where the templates are stored
+                logger_level: string or logging variable
+                    Set verbosity level for logger (default: logging.NOTSET)
+                """
+
+        self.__detector_file = detector_file
+
+        self.logger.setLevel(logger_level)
+
+        # set up the efield to voltage converter
+        self.__efieldToVoltageConverter.begin(debug=False)
+
+        self.__debug = debug
+
+        self.__template_save_path = template_save_path
+
+    def set_template_parameter(self, template_run_id=[0, 0, 0], template_event_id=[0, 1, 2], template_station_id=[101, 101, 101], template_channel_id=[0, 0, 0], Efield_width=[5, 4, 2],
+                                antenna_rotation=[160, 160, 160], Efield_amplitudes=[-0.2, 0.8], cr_zenith=[55, 55, 55], cr_azimuth=[0, 0, 0],
+                                sampling_rate=3.2 * units.GHz, number_of_samples=2048):
+        """
+        set_parameter_templates method
+
+        sets the parameter to create the template set
+
+        Parameters
+        ----------
+        template_run_id: list of int
```

Comment:
I think the documentation should comment on the default values. In particular the ones where it is != 0

Reply:
I added the default values